benches/search_comparison.rs

```diff
@@ -235,9 +235,7 @@ fn construction_bench_case<const MAX: usize, T, Coll>(
         for e in v.iter_mut() {
             r = r.wrapping_mul(1664525).wrapping_add(1013904223);
             if duplicates {
-                let r = (r % size) / 16 * 16;
-                let r = std::cmp::min(r, MAX);
-                *e = T::try_from(r).unwrap();
+                *e = T::try_from(r % 256).unwrap();
```

Comment:
I don't think this solution is correct. The `/ 16 * 16` part is correct IMO – it is required to provide duplicates (maybe it would be more concise to write as clearing LSB ` & ~0xF`). I think we need to modulo divide by `MAX` which is supposed to be the maximum value of the generated element in the collection.

Reply:
I am sorry I don't really understand this code 😅 I am going in blind and just doing what @jonhoo or you are saying.

I fixed the code according to your comment, please take a look

Reply:
Don't worry, I don't understand it fully either 😅. I'm pretty sure we will be able to find a working solution together.

Speaking of uncertainty. It's indeed somewhat hard to reason about what numbers does this code produce. What do you think of isolating the random number generating code in the separate iterator, so it can be tested and easily reused?

Reply:
I refactored it to separate function, but I am not sure if you want it to be put in the `lib.rs` so it could be present in the unit tests? Code under `benches` directory cannot be part of unit test.

Reply:
I noticed that code before had some unused trait bounds. And also there were logically dead code (see my comment in the PR) which I also removed.

Reply:
Also tests under `"Construction (with duplicates)"` were passing wrong flag to the bench function